From ec3c5660e67c122e2d5eb9cfa838c9709fccf8e0 Mon Sep 17 00:00:00 2001
From: Jason Martin <publicmsu@gmail.com>
Date: Fri, 3 Jun 2011 00:12:51 -0400
Subject: [PATCH 10/17] lirc_atiusb: fix buffer alloc to work with new kfifo

I came across posts with regards to lirc_atiusb and its compatibility
with newer kernels, and their use of kfifo.I spent a little bit of time
this evening, and was able to correct the issues with the driver, and
now have a functioning lirc_atiusb device under the 2.6.38 kernel.

Signed-off-by: Jarod Wilson <jarod@redhat.com>
---
 drivers/lirc_atiusb/lirc_atiusb.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/lirc_atiusb/lirc_atiusb.c b/drivers/lirc_atiusb/lirc_atiusb.c
index 52080de..bf2b631 100644
--- a/drivers/lirc_atiusb/lirc_atiusb.c
+++ b/drivers/lirc_atiusb/lirc_atiusb.c
@@ -1048,7 +1048,7 @@ static struct atirf_dev *new_irctl(struct usb_interface *intf)
 		goto new_irctl_failure_check;
 	}
 
-	if (lirc_buffer_init(driver->rbuf, dclen, 1)) {
+	if (lirc_buffer_init(driver->rbuf, dclen, 2)) {
 		mem_failure = 4;
 		goto new_irctl_failure_check;
 	}
-- 
1.8.3.1

